src/mpl/configure.ac

```diff
@@ -785,6 +785,78 @@ fi
 ## END OF PTHREAD MUTEX COMMON CHECK
 #######################################################################
 
+#######################################################################
+## START OF ATOMIC CODE
+#######################################################################
+
+PAC_ARG_ATOMIC_PRIMITIVES
+
+AC_DEFUN([MPL_ATOMIC_TEST_PROGRAM], [AC_LANG_PROGRAM([[
+        #include "$1"
+    ]],[[
+        MPL_atomic_int_t a, b;
+        int c;
+
+        MPL_atomic_relaxed_store_int(&a, 0);
+        MPL_atomic_relaxed_store_int(&b, 1);
+        c = MPL_atomic_relaxed_load_int(&a);
+
+        MPL_atomic_release_store_int(&a, 0);
+        MPL_atomic_release_store_int(&b, 1);
+        c = MPL_atomic_acquire_load_int(&a);
+
+        MPL_atomic_fetch_add_int(&a, 10);
+        MPL_atomic_fetch_sub_int(&a, 10);
+
+        c = MPL_atomic_cas_int(&a, 10, 11);
+        c = MPL_atomic_swap_int(&a, 10);
+
+        MPL_atomic_write_barrier();
+        MPL_atomic_read_barrier();
+        MPL_atomic_read_write_barrier();
+    ]])]
+)
+
+dnl MPL_TRY_ATOMIC_HEADER([header file from src/ dir],[symbolic name in configure],[HAVE_ macro suffix],[feature description])
+dnl Does an AC_LINK_IFELSE() to see if the header file works.
+AC_DEFUN([MPL_TRY_ATOMIC_HEADER],[
+if test "$with_mpl_atomic_primitives" = "auto" || \
+   test "$with_mpl_atomic_primitives" = "$2"
+then
```

Comment:
Personally I would think `with_atomic` is clear enough. At least I don't think we need `_mpl_` there.

Reply:
I agree; `_mpl_` is unnecessary. For now, I put `_mpl_` to avoid the duplicated definition; right now OpenPA (i.e., the current MPICH) uses `with_atomic_primitives` to choose the atomic primitives.

Reply:
I will remove it and in the future rename it to `--with_atomic_primitives` (e.g., https://github.com/pmodels/mpich/pull/3926/commits/93bf221534f8a9145fe80a6f66744add0ea399bc)